lib/src/core/engine.dart

```diff
@@ -433,9 +373,19 @@ class Engine extends Disposable with EventsEmittable<EngineEvent> {
       final isPrimaryOrPublisher = event.isPrimary ||
           (_hasPublished && event is EnginePublisherPeerStateUpdatedEvent);
 
-      if (isPrimaryOrPublisher && event.state.isDisconnectedOrFailed()) {
+      if (isPrimaryOrPublisher &&
```

Comment:
I'm not sure if `isPrimaryOrPublisher` is the right logic. if either peer connection is disconnected, I think we should trigger a handleDisconnect sequence.

the handling could be much simpler.

Reply:
resolved https://github.com/livekit/client-sdk-flutter/pull/194/commits/430bfa067cd6f8644973c2aa6701dadb01c08fe3